src/main/java/ie/ait/bteam/drcare/data/model/GeneralPractitioner.java

```diff
@@ -0,0 +1,9 @@
+package ie.ait.bteam.drcare.data.model;
+
+import javax.persistence.Entity;
+
+@Entity
+public class GeneralPractitioner extends User {
```

Comment:
I don't see the need for this new entity, as all the GP attributes are still in User. Better remove it, as UserValidation can check the GP values and when adding security, we don't have to check diff entities.